aiohttp/connector.py

```diff
@@ -770,6 +778,9 @@ def __init__(
         self._throttle_dns_events: Dict[Tuple[str, int], EventResultOrError] = {}
         self._family = family
         self._local_addr = local_addr
+        self._network_interface = network_interface
+        if self._network_interface and not SOCKET_SUPPORTS_BINDING_TO_DEVICE:
+            raise RuntimeError("Binding to interface is not supported by your OS.")
```

Comment:
I think it's best to avoid referring to a “you” in exceptions. It's not clear who that is and how they relate to the app being run. Maybe this would be more accurate?
```suggestion
            raise RuntimeError("Binding to interface is not supported by the current runtime (kernel or OS).")
```